analyses/transcriptomic-dimension-reduction/02-dimension-reduction.sh

```diff
@@ -0,0 +1,83 @@
+# Bethell and Taroni for CCDL 2019
+# Run dimension reduction for all subsets of gene expression for both methods:
+# RSEM and kallisto
+# 
+# Usage: bash 02-dimension-reduction.sh
+
+# This script should always run as if it were being called from
+# the directory it lives in.
+script_directory="$(perl -e 'use File::Basename;
+  use Cwd "abs_path";
+  print dirname(abs_path(@ARGV[0]));' -- "$0")"
+cd "$script_directory" || exit
+
+SEED=2019
+PERPLEXITY=10
+NEIGHBORS=15
+COUNT_THRESHOLD=100
+METADATA="../../data/pbta-histologies.tsv"
+OUTPUT="results"
+
+#### RSEM ----------------------------------------------------------------------
+
+Rscript --vanilla scripts/run-dimension-reduction.R \
+  --expression ../../data/pbta-gene-expression-rsem.fpkm.rds \
+  --metadata ${METADATA} \
+  --filename_lead rsem_all \
```

Comment:
Why set `filename_lead` and `seed` as command line options, rather than simply setting them in the R script where the function is executed?

Reply:
Having `seed` as a command line option allows us to easily run `run-dimension-reduction.R` multiple times with different seeds if we wanted to see how the t-SNE and UMAP (IIRC) results varied as a result of different seed values. `filename_lead` comes down to a similar reason as stated above:
> Avoiding hardcoding things like this give us more flexibility if we were to do something like add a new gene expression matrix from another method.

Reply:
Noted 👍 This is what I had in mind but I wanted to make sure that there was no other underlying reason.